src/hapticfeedback.h

```diff
@@ -0,0 +1,62 @@
+/* antimicrox Gamepad to KB+M event mapper
+ * Copyright (C) 2022 Max Maisel <max.maisel@posteo.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <QObject>
+#include <QString>
+#include <SDL2/SDL_gamecontroller.h>
+#include <stdint.h>
+
+#include "hapticfeedbackmode.h"
+
+/**
+ * @brief Represents a single haptic feedback effect on a PS5 controller trigger.
+ */
+class HapticFeedback : QObject
+{
+  public:
+    explicit HapticFeedback(QObject *parent, HapticFeedbackMode mode = HAPTIC_FEEDBACK_NONE, int strength = 0, int start = 0,
+                            int end = 0);
+
+    HapticFeedbackMode get_mode() const;
+    bool set_effect_mode(HapticFeedbackMode mode);
+    bool set_effect(int strength, int start, int end);
+
+    static void send(SDL_GameController *controller, const HapticFeedback &left, const HapticFeedback &right);
+    static HapticFeedbackMode from_string(const QString &mode);
+    static QString to_string(HapticFeedbackMode mode);
+
+  private:
+    enum
+    {
+        EFFECT_LEFT_EN = 0x04,
+        EFFECT_RIGHT_EN = 0x08,
+        LEGACY_RUMBLE_LEFT_EN = 0x0001,
+        LEGACY_RUMBLE_RIGHT_EN = 0x0002,
+        MIC_LIGHT_EN = 0x0100,
+
+        MIC_LIGHT_OFF = 0x00,
+        MIC_LIGHT_SOLID = 0x01,
+        MIC_LIGHT_PULSE = 0x02
+    };
```

Comment:
I think it would be good to provide link to source, where these values (and way of implementing this feature for PS5) can be found.

Reply:
What does `MIC_LIGHT_` means? microphone light?

Reply:
I added a link to a good source and changed it to use the recommended messages.

MIC_LIGHT is the microphone light. I changed it as well.